zerver/lib/pg_search.py

```diff
@@ -0,0 +1,48 @@
+import html
+import re
+
+
+def search_operand_to_tantivy_query(operand: str) -> str | None:
+    # TODO: Implement better parsing here so that we can support
+    # more complex queries; consider exposing some of the tantivy
+    # query language.
+    # A basic case that should at least be supported is search with phrase
+    # terms and word terms such as
+    # word1 "phrase with spaces" word2 "another phrase"
+    # For now this only supports:
+    # 1) single phrase search: "phrase with spaces"
+    # 2) words search: word1 word2 word3
```

Comment:
There might be a prep PR to consider in changing how we handle search operands in `add_narrow_conditions` to not squash them together there, and instead have them be separated for this code path. Not sure if clients send them in a way where that'd be helpful; I think that's a protocol issue that maybe got lost from @evykassirer's post-search-transition TODO list -- offering a UI for asking for `word1 word2 "consecutive phrase"` that actually gets sent through and processed properly by the server.
```
    if search_operands:                                                                                 
        is_search = True                                                                                
        query = query.add_columns(topic_column_sa(), column("rendered_content", Text))                  
        search_term = NarrowParameter(                                                                  
            operator="search",                                                                          
            operand=" ".join(search_operands),                                                          
        )                                                                                               
        query = builder.add_term(query, search_term)                                                    
```

Reply:
I just started a [thread for search followups on CZO](https://chat.zulip.org/#narrow/stream/101-design/topic/search.20followups), thanks for the reminder!

Reply:
I'm actually not sure if I'm unaware of some things that might become `search_operands", but searching for the expression
```
"this is a phrase" word1 word2
```

the `search_operands` the server sees in this codepath is:
```
> /srv/zulip/zerver/lib/narrow.py(1097)add_narrow_conditions()
-> search_term = NarrowParameter(
(Pdb) search_operands
['"this is a phrase" word1 word2']
```

So it's just one element and there's nothing to even `" ".join()`. If that's the case and the backend currently only ever just gets a single, big string, then I think it might be fine to leave it as it is (aside of cleaning up the confusing code that makes it like there's a real list involved)? It seems sensible that it would be the backend's job to parse the search query string?